code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json

```diff
@@ -12,8 +12,7 @@
         },
         "vpcId": {
             "type": "string",
```

Comment:
We've been taking the approach of only requiring the fields minimally required by docker-machine, Everything else, we allow machine to validate and then we propagate the error back up to the user.

Adding what you're suggeting isnt a bad idea though...especially as the drivers mature and stabalize.

Would you be comfortable with doing that in a separate PR further down the road?